### ISSUE TITLE
Clipboard Copy/Paste Do Not Works After a System Restart

### ISSUE THREAD

**Issue by rat-moonshine** [airsdk/Adobe-Runtime-Support#3039]
## Problem Description

AIR SDK: 50.2
Platform: macOS Ventura

Recently, we noticed that after a system restart, `Clipboard` copy/paste do not works _unless_ a particular way. 

`Event.COPY` is correctly dispatched by AIR. A call to `Clipboard.generalClipboard.setData()` returns _true_, indicating that it should have worked correctly. However, a call to `Clipboard.generalClipboard.getData()` returns _null_ instead of the text that we set.

## Steps to Reproduce

We are attaching herewith a sample ActionScript project that does following basic things:
- On user interaction, set some texts to Clipboard
- Display the Clipboard texts on screen using `Clipboard.generalClipboard.getData()`

To use the example, please try the following steps:

- Compile AIR captive runtime .app file from code attached
- Restart system **(IMPORTANT)**
- Launch the .app file
- Click on the red rectangle
- Press Cmd+C keyboard short cut

Results:

The TextField will display something like this:
```
[Event(type="copy" bubbles=true cancelable=true eventPhase=3)]
null
```

It should display this instead:
```
[Event(type="copy" bubbles=true cancelable=true eventPhase=3)]
hello
```

## Known Workarounds

Switch to another app, copy some text, and then switch back to the AIR app and try to copy again, everything works correctly.

## Attached Files

Example source project:
[CopyPasteAIRTest.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/14080377/CopyPasteAIRTest.zip)

Pre-compiled .app file:
[CopyPasteAIRTest.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/14080380/CopyPasteAIRTest.zip)

**Comment by piotrzarzycki21** [comment]
Just notice, that I don't necessary have to restart my system to reproduce this issue using above example. 

MacOs: Sonoma 14.3

**Comment by ajwfrost** [comment]
Thanks for following up on this one -> it seems like the behavior on macOS of one of their APIs is not quite what the documentation says, but equally, the code seems to have been written to the behaviour (most of the time) rather than to the API definition. So, making the code handle both of these, it should then cope with this scenario....

Will get this into a forthcoming release..

thanks

**Comment by johnwu** [comment]
I am still seeing this issue with MacOS 26.5 and SDK 51.1

**Comment by piotrzarzycki21** [comment]
> I am still seeing this issue with MacOS 26.5 and SDK 51.1

I don't think it was fixed at all, despite we raised it years ago trough support.

**Comment by ajwfrost** [comment]
Hi @johnwu @piotrzarzycki21 

We had put a fix for this into 51.0.1.3 for this .. see https://airsdk.harman.com/release_notes?q=3039

I've just tried the test case from the original post above, and could not get it to reproduce. Compiled the source and then created a (signed) .app bundle using the latest SDK, rebooted the machine and then ran it. Attached is the signed app in case you want to try it there?

[TestBundle.zip](https://github.com/user-attachments/files/29574058/TestBundle.zip)

thanks

**Comment by johnwu** [comment]
I have run the test app on MacOS Tahoe 26.5.1, and the only keys I can see it hears are copy and quit. I do not see any notification for paste or select all

**Comment by ajwfrost** [comment]
Hi @johnwu  -> if you mean the test app per the original issue above, that only has a 'copy' event handler in AS3.
If you're in a text field, then the normal macOS shortcuts should work for copy/paste/etc - but the text field in that example cannot be selected and is not editable.

thanks

**Comment by johnwu** [comment]
Could you include the source code for the test app so I can compare my code to it?
